### PR TITLE
test: e2e testing final

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-1236

### CODE REVIEW

File: tests/app/app.go [hunk 1]
```diff
@@ -508,6 +508,9 @@ func NewSimApp(
 	// set denom resolver to test variant.
 	app.FeeMarketKeeper.SetDenomResolver(&feemarkettypes.TestDenomResolver{})
 
+	// set denom resolver to test variant.
+	app.FeeMarketKeeper.SetDenomResolver(&feemarkettypes.TestDenomResolver{})
```
Comment (nivasan1) [inline tests/app/app.go:512]: WHy do this twice?

File: tests/e2e/suite.go [hunk 2]
```diff
@@ -153,33 +157,196 @@ func (s *TestSuite) TestQueryBaseFee() {
...
+					1,
+				)
+				s.Require().NoError(err, txResp)
+				s.Require().Equal(uint32(0), txResp.CheckTx.Code, txResp.CheckTx)
+				s.Require().Equal(uint32(0), txResp.TxResult.Code, txResp.TxResult)
+			}()
+
+			go func() {
+				defer wg.Done()
+				txResp, err := s.SendCoinsMultiBroadcast(
```
Comment (nivasan1) [inline tests/e2e/suite.go:202]: nit: move this logic to helper?

File: tests/e2e/suite.go [hunk 3]
```diff
@@ -153,33 +157,196 @@ func (s *TestSuite) TestQueryBaseFee() {
...
+				s.Require().NoError(err, txResp)
+				s.Require().Equal(uint32(0), txResp.CheckTx.Code, txResp.CheckTx)
+				s.Require().Equal(uint32(0), txResp.TxResult.Code, txResp.TxResult)
+			}()
+
+			wg.Wait()
+			fee := s.QueryBaseFee()
+			s.T().Log("base fee", fee.String())
+
+			if fee.AmountOf(feemarkettypes.DefaultFeeDenom).Equal(params.MinBaseFee) {
```
Comment (nivasan1) [inline tests/e2e/suite.go:236]: What does this check do?

File: tests/e2e/suite.go [hunk 4]
```diff
@@ -153,33 +157,196 @@ func (s *TestSuite) TestQueryBaseFee() {
...
+		height, err := s.chain.(*cosmos.CosmosChain).Height(context.Background())
+		s.Require().NoError(err)
+		s.WaitForHeight(s.chain.(*cosmos.CosmosChain), height+5)
+
+		fee := s.QueryBaseFee()
+		s.T().Log("base fee", fee.String())
+
+		amt, err := s.chain.GetBalance(context.Background(), s.user1.FormattedAddress(), baseFee[0].Denom)
+		s.Require().NoError(err)
+		s.Require().True(amt.LT(math.NewInt(initBalance)), amt)
```
Comment (nivasan1) [inline tests/e2e/suite.go:349]: Looks like we're checking the balance of user1's account? Are we not verifying properties of the fee-state here as well?

File: x/feemarket/ante/fee.go [hunk 5]
```diff
@@ -54,6 +54,9 @@ func (dfd FeeMarketCheckDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 	gas := feeTx.GetGas() // use provided gas limit
 
 	if len(feeCoins) != 1 {
+		if len(feeCoins) == 0 {
```
Comment (nivasan1) [inline x/feemarket/ante/fee.go:57]: nit: can we not name this `ErrTest`? Seems like it shld be `ErrNoFee` or smth?

File: tests/e2e/suite.go [hunk 6]
```diff
@@ -153,33 +157,196 @@ func (s *TestSuite) TestQueryBaseFee() {
 	})
 }
 
-func (s *TestSuite) TestSendTxUpdating() {
-	ctx := context.Background()
+func (s *TestSuite) TestSendTxDecrease() {
```
Comment (nivasan1) [inline tests/e2e/suite.go:160]: nit: would be useful to know what each of these tests are testing?

File: tests/e2e/suite.go (hunk 4 above)
Comment (aljo242) [reply tests/e2e/suite.go:349]: We are already checking the movement of the fee state in the for loop

File: x/feemarket/ante/fee.go (hunk 5 above)
Comment (aljo242) [reply x/feemarket/ante/fee.go:57]: OOPS this was a test debug that made it in here

File: tests/e2e/suite.go (hunk 3 above)
Comment (aljo242) [reply tests/e2e/suite.go:236]: we are checking that the fee has decreased and been "clamped" to the min base fee

**Pull request merged**